src/GeositeFramework/plugins/layer_selector_v2/WmsService.js

```diff
@@ -0,0 +1,157 @@
+define([
+        "dojo/_base/declare",
+        "dojo/Deferred",
+        "underscore",
+        "./lib/xmlToJson",
+        "framework/util/ajax",
+        "./util",
+        "./LayerNode",
+    ],
+    function(declare,
+             Deferred,
+             _,
+             xmlToJson,
+             ajaxUtil,
+             util,
+             LayerNode) {
+        "use strict";
+
+        var jsonResponseCache = [];
+
+        function get(url) {
+            return jsonResponseCache[url];
+        }
+
+        return declare(null, {
+            constructor: function(server) {
+                this.server = server;
+                this.defaultAjaxOptions = {
+                    format: 'text',
+                    content: ''
+                };
+            },
+
+            getServiceUrl: function() {
+                return util.urljoin(this._getBaseServiceUrl() + '?request=GetCapabilities&service=WMS');
+            },
+
+            _getBaseServiceUrl: function() {
+                // If name exists, this is likely an AGS WMS service,
+                // as opposed to GeoServer.
+                if (this.server.name) {
+                    return util.urljoin(this.server.url, this.server.name, 'MapServer/WMSServer');
+                } else {
+                    return this.server.url;
+                }
+            },
+
+            // Return a promise containing map service data.
+            fetchMapService: function() {
+                var self = this;
+                return ajaxUtil.fetch(this.getServiceUrl(), this.defaultAjaxOptions).
+                        then(function(response) {
+                            jsonResponseCache[self.getServiceUrl()] = self._transformServiceData(response);
+                            return self.responseJSON;
+                        });
+            },
+
+            // Return cached map service data.
+            getServiceData: function() {
+                return get(this.getServiceUrl());
+            },
+
+            // Return a promise with service layer data.
+            fetchLayerDetails: function(tree, layerId) {
+                // There are no layer details for a
+                // WMS layer.
+                return new Deferred().resolve(function() {
+                    return {};
+                });
```

Comment:
I don't know why dojo let's you do this but I think we should always resolve a deferred with a value instead of a function.